index.js

```diff
@@ -303,43 +324,86 @@ FastForEach.prototype.insertAllAfter = function(nodeOrNodeArrayToInsert, insertA
   return nodeOrNodeArrayToInsert;
 };
 
+// checks if the deleted data item should be handled with delay for a possible reuse at additions
+FastForEach.prototype.shouldDelayDeletion = function (data) {
+  return this.allowMoveNodes && (typeof data === "object" || typeof data === "function");
+}
+
+// get the cached index from the data item which points to its pendind deletion info
+FastForEach.prototype.getPenndingDeleteIndex = function (data) {
```

Comment:
Hm.. it's just kind of SoC, I thought that the logic which extracts info from a data item could be broken out to a small helper function. Probably someone will want to override it, etc.